open-banking-accelerator/internal-apis/internal-webapps/com.wso2.openbanking.accelerator.event.notifications.endpoint/src/main/java/com/wso2/openbanking/accelerator/event/notifications/endpoint/util/EventSubscriptionUtils.java

```diff
@@ -74,34 +72,16 @@ public static JSONObject getJSONObjectPayload(HttpServletRequest request) throws
      * @return Response
      */
     public static Response mapEventSubscriptionServiceResponse(EventSubscriptionResponse eventSubscriptionResponse) {
-        String status = eventSubscriptionResponse.getStatus();
+        int status = eventSubscriptionResponse.getStatus();
         if (eventSubscriptionResponse.getErrorResponse() == null) {
-            switch (status) {
-                case EventNotificationConstants.CREATED:
-                    return Response.status(Response.Status.CREATED)
-                            .entity(eventSubscriptionResponse.getResponseBody())
-                            .build();
-                case EventNotificationConstants.OK:
-                    return Response.status(Response.Status.OK)
-                            .entity(eventSubscriptionResponse.getResponseBody())
-                            .build();
-                case EventNotificationConstants.NO_CONTENT:
-                    return Response.status(Response.Status.NO_CONTENT)
-                            .entity(eventSubscriptionResponse.getResponseBody())
-                            .build();
-            }
-        } else if (status.equals(EventNotificationConstants.BAD_REQUEST)) {
-            return Response.status(Response.Status.BAD_REQUEST)
-                    .entity(EventNotificationUtils.getErrorDTO(
-                            EventNotificationEndPointConstants.INVALID_REQUEST,
-                            eventSubscriptionResponse.getErrorResponse()))
+            return Response.status(status)
+                    .entity(eventSubscriptionResponse.getResponseBody())
```

Comment:
Won't this throw a NPE?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/3/commits/a31eef6176e5003bd359e183aae45a90a5874f2b